src/main/java/io/elastic/jdbc/Utils.java

```diff
@@ -1,75 +1,307 @@
-        return "";
+      } else {
+        if (body.get(colName) != null) {
+          statement.setString(paramNumber, body.getString(colName));
+        } else {
+          statement.setNull(paramNumber, Types.VARCHAR);
+        }
+      }
+    } catch (java.lang.NumberFormatException e) {
+      String message = String
+              .format("Provided data: %s can't be cast to the column %s datatype",body.get(colName),
+                      colName);
+      throw new RuntimeException(message);
+    }
+  }
+
+  private static String detectColumnType(Integer sqlType) {
+    if (sqlType == Types.NUMERIC || sqlType == Types.DECIMAL || sqlType == Types.TINYINT
+        || sqlType == Types.SMALLINT || sqlType == Types.INTEGER || sqlType == Types.BIGINT
+        || sqlType == Types.REAL || sqlType == Types.FLOAT || sqlType == Types.DOUBLE) {
+      return "number";
+    }
+    if (sqlType == Types.TIMESTAMP) {
+      return "timestamp";
+    }
+    if (sqlType == Types.DATE) {
+      return "date";
     }
+    if (sqlType == Types.BIT || sqlType == Types.BOOLEAN) {
+      return "boolean";
+    }
+    return "string";
+  }
+
+  private static String getColumnType(String columnName) {
+    return columnTypes.get(columnName.toLowerCase());
+  }
+
+  private static boolean isNumeric(String columnName) {
+    String type = getColumnType(columnName);
+    return type != null && type.equals("number");
+  }
+
+  private static boolean isTimestamp(String columnName) {
+    String type = getColumnType(columnName);
+    return type != null && type.equals("timestamp");
+  }
 
-    private static Integer getPort(final JsonObject config, final Engines engineType) {
-        final JsonElement value = config.get(CFG_PORT);
-        if (value != null && !value.isJsonNull() && !value.getAsString().isEmpty()) {
-            return value.getAsInt();
+  private static boolean isDate(String columnName) {
+    String type = getColumnType(columnName);
+    return type != null && type.equals("date");
+  }
+
+  private static boolean isBoolean(String columnName) {
+    String type = getColumnType(columnName);
+    return type != null && type.equals("boolean");
+  }
+
+  public static Map<String, String> getColumnTypes(Connection connection, Boolean isOracle,
+      String tableName) {
+    DatabaseMetaData md;
+    ResultSet rs = null;
+    Map<String, String> columnTypes = new HashMap<String, String>();
+    String schemaName = null;
+    try {
+      md = connection.getMetaData();
+      if (tableName.contains(".")) {
+        schemaName = tableName.split("\\.")[0];
+        tableName = tableName.split("\\.")[1];
+      }
+      rs = md.getColumns(null, schemaName, tableName, "%");
+      while (rs.next()) {
+        String name = rs.getString("COLUMN_NAME").toLowerCase();
+        String type = detectColumnType(rs.getInt("DATA_TYPE"));
+        columnTypes.put(name, type);
+      }
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    } finally {
```

Comment:
should we close rs in finally block?
May be we can move it above and remove catch block?

Reply:
we should (was discussed)